tensorflow/core/kernels/mkl/mkl_conv_ops.cc

```diff
@@ -2199,83 +2358,141 @@ REGISTER_MKL_KERNEL("QuantizedDepthwiseConv2DWithBias", NoOp, quint8, float,
+#undef SUMMAND_TYPE_CONSTRAINT
 #undef BIAS_TYPE_CONSTRAINT
 #undef TEMPLATE_ARGS
 #undef LABEL
 
-#define LABEL .Label(mkl_op_registry::kMklQuantizedOpLabel)
 #define TEMPLATE_ARGS(CPUDevice, input_type, bias_type, output_type, \
-                      accu_type, has_bias, is_depthwise, is_native)  \
-<CPUDevice, input_type, bias_type, output_type, accu_type, has_bias, \
-      is_depthwise, is_native>
+                      summand_type, is_depthwise, legacy_fused_ops,  \
+                      num_fused_ops)                                 \
+<CPUDevice, input_type, bias_type, output_type, summand_type, is_depthwise, legacy_fused_ops, num_fused_ops>
 #define BIAS_TYPE_CONSTRAINT(bias_type)
-REGISTER_MKL_KERNEL_ALL_INPUT_TYPES("_MklQuantizedConv2D", MklQuantizedConv2DOp,
-                                    float, qint32, qint32, false, false, true);
-REGISTER_MKL_KERNEL("_MklQuantizedConv2DPerChannel", MklQuantizedConv2DOp,
-                    quint8, float, qint32, qint32, false, false, true);
+#define SUMMAND_TYPE_CONSTRAINT(summand_type)
+#define LABEL .Label(mkl_op_registry::kMklQuantizedOpLabel)
+REGISTER_MKL_KERNEL_ALL_INPUT_TYPES("_MklQuantizedConv2D", MklQuantizedConvOp,
+                                    float, qint32, qint32, false,
+                                    quantized_fusions::none, 0);
+REGISTER_MKL_KERNEL_ALL_INPUT_TYPES("_MklQuantizedConv2DPerChannel",
+                                    MklQuantizedConvOp, float, qint32, qint32,
+                                    false, quantized_fusions::none, 0);
 REGISTER_MKL_KERNEL_ALL_INPUT_TYPES("_MklQuantizedConv2DWithBias",
-                                    MklQuantizedConv2DOp, float, qint32, qint32,
-                                    true, false, true);
+                                    MklQuantizedConvOp, float, qint32, qint32,
+                                    false, quantized_fusions::bias, 1);
 REGISTER_MKL_KERNEL_ALL_INPUT_TYPES("_MklQuantizedConv2DWithBiasAndRelu",
-                                    MklQuantizedConv2DReluOp, float, qint32,
-                                    qint32, true, false, true);
-REGISTER_MKL_KERNEL("_MklQuantizedConv2DWithBiasSumAndRelu",
-                    MklQuantizedConv2DSumReluOp, quint8, float, qint32, qint32,
-                    true, false, true);
-REGISTER_MKL_KERNEL("_MklQuantizedConv2DAndRequantize", MklQuantizedConv2DOp,
-                    quint8, float, qint8, qint8, false, false, true);
-REGISTER_MKL_KERNEL("_MklQuantizedConv2DAndRelu", MklQuantizedConv2DReluOp,
-                    quint8, float, qint32, qint32, false, false, true);
+                                    MklQuantizedConvOp, float, qint32, qint32,
+                                    false, quantized_fusions::bias_relu, 2);
+REGISTER_MKL_KERNEL("_MklQuantizedConv2DWithBiasSumAndRelu", MklQuantizedConvOp,
+                    quint8, float, qint32, qint32, false,
+                    quantized_fusions::bias_sum_relu, 3);
```

Comment:
I don't think we need the vector of string (for fused op names) and vector size as the last two template parameters. Since the old API doesn't support permutations of ops (e.g., only bias + sum + relu, never relu + bias + sum or other orders), we can just use the 64-bit flag as just one template parameter instead. For example, the registration here would be:
```c++
REGISTER_MKL_KERNEL("_MklQuantizedConv2DWithBiasSumAndRelu", MklQuantizedConvOp,
                    quint8, float, qint32, qint32, false, kBias | kSum | kRelu);
```
And this number can be stored right away as `fused_op_flags_` in the class.

Reply:
As I mentioned above, can this be done once the old API is removed? Please let me know what you think.

Reply:
Postponing to after the branch cut sounds good to me.

Reply:
@penpornk Although the old API doesn't support permutations of ops, the ordering of `sum` and `activation` post-ops would be needed (as I mentioned above) since they are passed to oneDNN in the same order in which they were received.